govcd/vapptemplate.go

```diff
@@ -180,3 +180,86 @@ func (vAppTemplate *VAppTemplate) Delete() error {
 	}
 	return nil
 }
+
+// GetVAppTemplateByHref finds a vApp template by HREF
+// On success, returns a pointer to the vApp template structure and a nil error
+// On failure, returns a nil pointer and an error
+func (vcdClient *VCDClient) GetVAppTemplateByHref(href string) (*VAppTemplate, error) {
+	return getVAppTemplateByHref(&vcdClient.Client, href)
+}
+
+// GetVAppTemplateById finds a vApp Template by ID.
+// On success, returns a pointer to the VAppTemplate structure and a nil error.
+// On failure, returns a nil pointer and an error.
+func (vcdClient *VCDClient) GetVAppTemplateById(vAppTemplateId string) (*VAppTemplate, error) {
+	return getVAppTemplateById(&vcdClient.Client, vAppTemplateId)
+}
+
+// QueryVAppTemplateById Finds a vApp Template by its URN.
```

Comment:
As mentioned in [PR 931](https://github.com/vmware/terraform-provider-vcd/pull/931), we should decouple the retrieval of the vApp template from its sync state.

Especially important is that this function will return entity not found when the vApp template exists but it is not fully synchronized.

Reply:
Fixed. Now the errors are different.